packages/netsuite-adapter/src/filters/data_types_custom_fields.ts

```diff
@@ -124,10 +124,16 @@ const filterCreator: FilterCreator = ({ isPartial, elementsSourceIndex }) => ({
     const nameToType = _.keyBy(dataTypes, e => e.elemID.name)
 
     if (isPartial) {
+      const fetchedIds = new Set(elements
+        .filter(isInstanceElement)
+        .map(instance => instance.elemID.getFullName()))
+
       Object.entries((await elementsSourceIndex.getIndexes()).customFieldsIndex)
         .filter(([type]) => type in nameToType)
         .forEach(([type, fields]) => {
-          fields.forEach(field => addFieldToType(nameToType[type], field, nameToType))
+          fields
+            .filter(field => !fetchedIds.has(field.elemID.getFullName()))
```

Comment:
Am I missing something or it should be?
```suggestion
            .filter(field => fetchedIds.has(field.elemID.getFullName()))
```

Reply:
I want to use the field from the `elementSourceIndex` only if I did not fetch that field from the service in the current fetch, so I filter out all the fields that exist in `fetchedIds`

Fields that were fetched from the service in the current fetch are added to the type in https://github.com/salto-io/salto/pull/2352/files/238dfd19ab41049b5920c0cf33260fed10dfa17d#diff-66d91a05666309950a78bbc98a6a54711ba901b702c1958e0ed78885cf6f82bcR140

Reply:
Got it, makes sense. I missed the second part.
Could you please add a comment explaining it?